incubation.md

```diff
@@ -0,0 +1,79 @@
+* Enables the core development team to assess the readiness an applicability of proposed technologies against a backdrop of the broader project design efforts. 
+* Allows the core team to re-evaluate existing jupyter org repos to determine if any need to be moved to jupyter-incubator.
+
+## Proposal for incubation
+
+A subproject enter into incubation by submitting a proposal for incubation. This process is designed to be lightweight and quick:
+
+1. The proposers should fill out incubation application by sumitting a pull request to the
+   jupyter-incubator/proposals repository.
+2. The proposed subproject must have a mentor/sponsor/champion who is an active Steering 
+   Council member.
+3. The proposal for incubation will be discussed by the community and approved or rejected
+   by the consensus of the Steering Council.
+4. If approved the proposal PR will be merged, if rejected, it will be closed.
+
+## Incubation period
+
+Once a project has become an incubation subproject, the following steps will be taken:
+
+1. A GitHub repo will be created under the jupyter-incubator organization
+2. A GitHub team will created with read/write accees to the repo, including the
+   mentor/sponsor/champion.
+3. Standard Jupyter LICENSE file will be added to the repo.
+
+The goal of the incubation period is to demonstrate that the subproject is going to take off
+with an active developer and user community. There is no particular length of time required,
+but most projects will be in incubation for at least 6 months to 1 year.
+
+## Proposal for graduation from incubation
+
+When a incubating subproject graduates, it becomes an official part of Project Jupter and
+is moved to the jupyter GitHub organization.
+
+To graduate from the incubation period, a subproject should submit a second pull request
+against the jupyter-incubator/proposals repo that completes the second half of the incubator
+proposal.
+
+* Initiated by a second PR against the proposals repo that finishes the rest of the application.
+* Requires consensus of the Steering Council.
+* Three outcomes
+  - More incubation with recommendations
+  - Voluntary retiremet.
+  - Forced termination.
+  - Gradution as a new core project.
+  - Integration into existing core project.
+
+## Criteria for graduation
+
+The following criteria will be used in evaluating subprojects for graduation.
+
+* Active developer community that offers a sustainable model for future development.
+* Active user community.
+* Solid software engineering with docs on readthedocs, tests, Travis.
+* Demonstrates continuing growth and development.
+* Integration with the rest of the project.
+* Willingness to submit to the Jupyter governance model.
+* Well defined scope
+* Well packaged (pip, conda, npm, bower, docker)
+
+## Graduation
+
+* Transfer GitHub repo over to jupyter org.
+* Make team for the project.
+
+
+## Questions
+
+* Do we want to introduce the idea of a repository/subproject maintainer?
+* 
```

Comment:
Addressing [concerns from Google Group Post](https://groups.google.com/forum/#!topic/jupyter/V8wqB9_1TQw)

* How will the incubating projects be evaluated? Addressed in line 36:  Proposal for graduation from incubation

* What about situations where a few Steering Council members get excited about something and short circuits the incubator process and directly creates a main project repo? *Well governance is there for a reason, including adherence by council members. Excitement can be addressed by aiding in the creation of a justifiable proposal.*

* Who votes on approval for graduation? Steering Council? Community?  This is a process that will need to evolve. For starters, I suggest that if a proposal does not have any **red-flags** by any community members then the proposal should be allowed to be raised for a vote at a upcoming standup mtg. A **red-flag** would be raised within the proposal doc. Those in attendance at the scheduled standup mtg can vote (+/-1). This will require that such items needs to be communicated as part of the standup agenda (minimally a week in advance).
 
* Should we move some of the existing experimental jupyter projects to the incubator? I think this is a worthwhile discussion and exercise. However, it raises topics outside the incubator. For example, where we put jupyter/nature-demo? Is it a core asset for someone interested in using Project Jupiter? I think this exercise will identify a list of categories of existing repos. Before we answer what we do or where things go, we need to understand the types of repos we have to deal with and why they are associated with the core Jupiter Project. Such an exercise may also help us better articulate the criteria for graduating/approving a Jupiter repo proposal.